metalearners/metalearner.py

```diff
@@ -311,6 +348,16 @@ def _validate_models(self) -> None:
                 factory, predict_method, name=f"treatment model {model_kind}"
             )
 
+    @classmethod
+    def _validate_scoring(cls, scoring: Mapping[str, list[str | Callable]]):
+        if not set(scoring.keys()) <= (
+            set(cls.nuisance_model_specifications().keys())
+            | set(cls.treatment_model_specifications().keys())
+        ):
+            raise ValueError(
+                "scoring dict keys need to be a subset of the model names in the MetaLearner"
```

Comment:
```suggestion
                "keys of parameter scoring need to be a subset of MetaLearner's base model names"
```

Reply:
Why do we need them to be a subset? Why don't we ignore superficial ones?

E.g. if someone is iterating over various MetaLearners and is trying to reuse a scoring dict for several MetaLearners, this won't be possible even though we might just allow for it.

E.g.

```python
prefixes = ["T", "DR"]
for prefix in prefixes:
    metalearner = metalearner_factory(prefix)(
        nuisance_model_factory=LGBMRegressor,
        propensity_model_factory=LGBMClassifier,
        is_classification=False,
        n_variants=2,  
    )
    print(metalearners.evaluate(
        X=X,
        y=y,
        w=w,
        is_oos=False,
        scoring={"propensity_model": "nll", "variant_outcome_model": "rmse"},
    ))
```

Reply:
[0150106](https://github.com/Quantco/metalearners/pull/8/commits/01501067b288760b5db9b2857699671541a54a61)